fem/hyperbolic.cpp

```diff
@@ -33,15 +56,15 @@ void HyperbolicFormIntegrator::AssembleElementVector(const FiniteElement &el,
    // shape function value at an integration point
    Vector shape(dof);
    // derivative of shape function at an integration point
-   DenseMatrix dshape(dof, el.GetDim());
+   DenseMatrix dshape(dof, Tr.GetSpaceDim());
    // state value at an integration point
    Vector state(num_equations);
    // flux value at an integration point
    DenseMatrix flux(num_equations, el.GetDim());
```

Comment:
Also, this should be `Tr.GetSpaceDim()` if we want to use the physical derivative with `sdim != dim`.

Reply:
Not really, the velocity must be tangential to the manifold (to stay on the manifold). It should be this integrator, which handles the tangential integration or there could be something like `FluxFunction::ComputePhysFlux()`. Anyway, I would leave it for now.